### DISCUSSION TITLE
Failed run profiling python code that contain duckdb

### DISCUSSION THREAD

**Discussion by ibnubay** [plasma-umass/scalene#865] [General]
Env:
- Macos Sonoma(14.5)
- python 3.11.9
- scalene 1.5.44.1
- Command: scalene --html --outfile elt_duckdb_direct.html python elt_duckdb_direct.py

It failed to connect to remote DB (mysql) when start run scalene. Show error message below.
`duckdb.duckdb.IOException: IO Error: Failed to connect to MySQL database with parameters`
After html output it show, scalene command still running when I check for the process.

But when I run that python script as usual without scalene, no error connection occurred.
Any one have some suggestion or have same case like this?

Ty

**Comment by emeryberger** [comment]
Please try installing from the repo version: `pip install git+https://github.com/plasma-umass/scalene`; after that, see what happens if you specify `--cpu` (I doubt this is the issue but let's see).

**Reply by emeryberger** [reply]
You might also want to put `---` right before the Python script in your command line.

**Reply by ibnubay** [reply]
Hi @emeryberger , ty for your response.

I already install package from scalene repo as you suggest before and add option only cpu.
`scalene --cpu --- python elt_duckdb_direct.py`

And still have same error
`duckdb.duckdb.IOException: IO Error: Failed to connect to MySQL database with parameters`

**Reply by ibnubay** [reply]
I'm compare with psrecord, and success running.
`psrecord 'python elt_duckdb_direct.py' --include-children --plot plot-elt_duckdb_direct-parquet.png`

**Reply by ibnubay** [reply]
And I'm curios to test connect into "Postgresql" with command below with "success" running, but have stuck process running in background for a while and not memory info captured. It seem failed to capture profiling like "Mysql" before but no error connection show.
`scalene --memory-leak-detector --html --outfile elt_duckdb_direct.html --- python elt_duckdb_direct.py`
